src/app/helpers/worldgen.ts

```diff
@@ -641,6 +748,9 @@ export async function worldgenGenerateWorld(
     setWorldGenStatus(`Giving treasure to the world...`);
     fillSpacesWithLoot(nodes);
 
+    setWorldGenStatus(`Filling fog gaps...`);
+    fillFogGaps(config, nodes, rng, counts);
```

Comment:
this step needs to happen before giving treasure and guardians to the world so these nodes can get assigned elements, loot, and guardians.